mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -4640,6 +4655,40 @@ def _get_filter_clauses_for_search_traces(filter_string, session, dialect):
                 )
                 span_filters.append(span_subquery)
                 continue
+            elif SearchTraceUtils.is_feedback(key_type, key_name, comparator):
+                # Create subquery to find traces with matching feedback
+                # Filter by feedback name and check the value
+                feedback_subquery = (
+                    session.query(SqlAssessments.trace_id.label("request_id"))
+                    .filter(
+                        SqlAssessments.assessment_type == "feedback",
+                        SqlAssessments.name == key_name,
+                        SearchTraceUtils.get_sql_comparison_func(comparator, dialect)(
+                            SqlAssessments.value, value
+                        ),
+                    )
+                    .distinct()
+                    .subquery()
+                )
+                span_filters.append(feedback_subquery)
+                continue
+            elif SearchTraceUtils.is_expectation(key_type, key_name, comparator):
+                # Create subquery to find traces with matching expectation
+                # Filter by expectation name and check the value
+                expectation_subquery = (
+                    session.query(SqlAssessments.trace_id.label("request_id"))
+                    .filter(
+                        SqlAssessments.assessment_type == "expectation",
+                        SqlAssessments.name == key_name,
+                        SearchTraceUtils.get_sql_comparison_func(comparator, dialect)(
+                            SqlAssessments.value, value
+                        ),
+                    )
+                    .distinct()
+                    .subquery()
+                )
+                span_filters.append(expectation_subquery)
+                continue
```

Comment:
It seems the logic is identical except for assessent_type, can we combine the logic?